File: src/server/backend/runtime/plugins/llm_before/qwen3guard.py
"""Qwen3Guard plugin for LLM input events."""
from __future__ import annotations

from shared.schemas.events import EventType, RuntimeEvent

from backend.runtime.plugins.common.patterns import text_of
from backend.runtime.plugins.qwen3guard import Qwen3GuardPluginBase
from backend.runtime.plugins.registry import register


@register(
    name="qwen3guard_input",
    description="Classify LLM input with Qwen3Guard before model execution.",
)
class Qwen3GuardInputPlugin(Qwen3GuardPluginBase):
    event_types = [EventType.LLM_INPUT]
    policy_scope = "llm_input"

    def _messages_for_event(self, event: RuntimeEvent) -> list[dict[str, str]]:
        messages = []
        for item in event.payload.messages:
            role = str(item.get("role") or "user").strip() or "user"
            content = text_of(item.get("content"))
            if content:
                messages.append({"role": role, "content": content})
        return messages

File: src/client/js/agentguard/plugins/llm_before/qwen3guard.js
"use strict";

const { BasePlugin } = require("../base");
const { EventType } = require("../../schemas/events");
const { register } = require("../registry");
const { checkWithQwen3Guard, textOf } = require("../common/qwen3guard");

class Qwen3GuardInputPlugin extends BasePlugin {
  constructor(options = {}) {
    super(options);
    this.event_types = [EventType.LLM_INPUT];
    this.policy_scope = "llm_input";
  }

  check(event) {
    const messages = [];
    for (const item of event.payload?.messages || []) {
      const role = String(item?.role || "user").trim() || "user";
      const content = textOf(item?.content);
      if (content) {
        messages.push({ role, content });
      }
    }
    return checkWithQwen3Guard(this, messages, this.policy_scope);
  }
}

register(
  "qwen3guard_input",
  "Classify LLM input with Qwen3Guard before model execution.",
)(Qwen3GuardInputPlugin);

module.exports = {
  Qwen3GuardInputPlugin,
};
